### PR TITLE
Set up GitHub actions

### PR DESCRIPTION
**Pull request by Martin7-1**
This PR introduces Github Actions Set up, including:

1. Issue (including bug report and feature request) and PR template.
2. Main CI, which will build and check `IT`.
3. Release and snapshot release.

/cc @kpavlov Would you please help me review this PR?

/cc @langchain4j BTW, for realease/snapshot release and IT, Could I just use your key and secret directly? (including Testcontainers Cloud, Deploy GPG key and so on...)

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  java_build:
+    if: github.repository == 'langchain4j/langchain4j-community'
+    strategy:
+      matrix:
+        java_version:
+          - 17
+          - 21
+        include:
+          - java_version: '17'
+            included_modules: ''
```
Comment (kpavlov) [inline .github/workflows/nightly.yaml:18]: maybe, rename it to `maven_opts`, because it can be anything

File: pom.xml [hunk 2]
```diff
@@ -26,9 +30,17 @@
...
 
         <!-- LangChain4j Community integration of model provider -->
+
     </modules>
 
     <build>
+        <extensions>
+            <extension>
+                <groupId>com.vackosar.gitflowincrementalbuilder</groupId>
+                <artifactId>gitflow-incremental-builder</artifactId>
```
Comment (kpavlov) [inline pom.xml:40]: @langchain4j Do we really need gf incremental builders here and in other projects, if it is immediately [disabled](https://github.com/langchain4j/langchain4j-community/pull/4/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R22)?

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ matrix.java_version }}
+          distribution: 'temurin'
+          cache: 'maven'
+
+      - name: Build with JDK ${{ matrix.java_version }}
+        run: |
+          ## compile and verify javadocs on ALL modules
+          mvn -B -U -T4C process-test-classes javadoc:aggregate
```
Comment (kpavlov) [inline .github/workflows/nightly.yaml:36]:
```suggestion
          mvn -B -U -T4C test javadoc:aggregate
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+name: Feature request
+about: Suggest an idea for this project
+title: "[FEATURE] "
+labels: enhancement
+assignees: ''
+
+---
+
+**Is your feature request related to a problem? Please describe.**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/feature_request.md:10]:
```suggestion
## Is your feature request related to a problem? Please describe.
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
+
+**Describe the solution you'd like**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/feature_request.md:13]:
```suggestion
## Describe the solution you'd like
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+<!-- A clear and concise description of what you want to happen. -->
+
+**Describe alternatives you've considered**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/feature_request.md:16]:
```suggestion
## Describe alternatives you've considered
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+<!-- A clear and concise description of any alternative solutions or features you've considered. -->
+
+**Additional context**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/feature_request.md:19]:
```suggestion
## Additional context
```

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+<!-- Please describe the changes you made. -->
+
+
+## General checklist
+<!-- Please double-check the following points and mark them like this: [X] -->
+- [ ] There are no breaking changes
+- [ ] I have added unit and integration tests for my change
+- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
+<!-- Before adding documentation and example(s) (below), please wait until the PR is reviewed and approved. -->
+- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
```
Comment (kpavlov) [inline .github/pull_request_template.md:27]:
```suggestion
- [ ] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [ ] I have manually run all the unit tests in _all_ modules, and they are all green
- [ ] I have manually run all integration tests in the module I have added/changed, and they are all green
<!-- Before adding documentation and example(s) (below), please wait until the PR is reviewed and approved. -->
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
```

**Review by kpavlov — COMMENTED** [review]
@Martin7-1 , thank you for the PR. I left few comments

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+about: Create a report to help us improve
+title: "[BUG] "
+labels: bug
+assignees: ''
+
+---
+
+<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->
+
+**Describe the bug**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/bug_report.md:12]:
```suggestion
## Describe the bug
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+<!-- A clear and concise description of what the bug is. -->
+
+**Log and Stack trace**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/bug_report.md:15]:
```suggestion
## Log and Stack trace
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+<!-- Please provide a log and a stack trace (with exception), if applicable. -->
+
+**Minimal Example to Reproduce**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]:
```suggestion
## Minimal Example to Reproduce
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+<!-- Please provide a relevant code snippets to reproduce this bug. -->
+
+**Expected behavior**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/bug_report.md:21]:
```suggestion
## Expected behavior
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+<!-- A clear and concise description of what you expected to happen. -->
+
+**Please complete the following information:**
+- LangChain4j Community version:
+- LangChain4j Community Component you used (e.g. `langchain4j-community-clickhouse`):
+- Java version: 
+- Spring Boot version (if applicable): 
+
+**Additional context**
```
Comment (kpavlov) [inline .github/ISSUE_TEMPLATE/bug_report.md:30]:
```suggestion
## Additional context
```

**Review by kpavlov — COMMENTED** [review]
@Martin7-1 , thank you for the PR. I left few comments

File: pom.xml (hunk 2 above)
Comment (langchain4j) [reply pom.xml:40]: @kpavlov perharps it makes no sense right now, but once there will be many modules, builds will be faster with GIB

File: .github/workflows/nightly.yaml (hunk 1 above)
Comment (Martin7-1) [reply .github/workflows/nightly.yaml]: Done.

File: .github/workflows/nightly.yaml (hunk 3 above)
Comment (Martin7-1) [reply .github/workflows/nightly.yaml]: Done.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 4 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Done.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 6 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Done.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 5 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Done.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 7 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Done.

File: .github/pull_request_template.md (hunk 8 above)
Comment (Martin7-1) [reply .github/pull_request_template.md]: Done.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 9 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Done.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 10 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Done.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 11 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Done.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 13 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Done.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 12 above)
Comment (Martin7-1) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Done.

**Review by kpavlov — APPROVED** [review]

**Pull request merged**